### DISCUSSION TITLE
can I run pwsh from an msbuild exec target?

### DISCUSSION THREAD

**Discussion by af4jm** [PowerShell/PowerShell#19562] [Q&A]
of these 3 lines, the first 2 work on Windows but are not cross-platform

the last one errors that it can't find "C:\bin\pwsh.dll" which makes no sense & raises the question of how do you make a cross-platform target to run a PowerShell script

``` xml
<Exec Command="powershell.exe -ExecutionPolicy Bypass -NoProfile -NonInteractive -Command &quot;&amp; { &amp;&apos;$(MSBuildProjectDirectory)\..\Scripts\Do-Something.ps1&apos; }&quot;" />
<Exec Command="&quot;C:\Program Files\PowerShell\7\pwsh.exe&quot; -ExecutionPolicy Bypass -NoProfile -NonInteractive -Command &quot;&amp; { &amp;&apos;$(MSBuildProjectDirectory)\..\Scripts\Do-Something.ps1&apos; }&quot;" />
<Exec Command="pwsh.exe -ExecutionPolicy Bypass -NoProfile -NonInteractive -Command &quot;&amp; { &amp;&apos;$(MSBuildProjectDirectory)\..\Scripts\Do-Something.ps1&apos; }&quot;" />
```

**Comment by rhubarb-geek-nz** [comment]
Running just 'pwsh' from the `PATH` should be enough. `pwsh` is not in the same place on macOS compared to Linux so PATH is the best way to find it, and it won't have the `.exe` extension.

**Reply by af4jm** [reply]
that's what I thought, but when I tried it in Windows, instead of finding pwsh.exe in the path, it gave me the bogus error that c:\bin\pwsh.dll doesn't exist... dlls don't
execute by themselves anyway... guess i need to find an msbuild place to ask :(

**Reply by af4jm** [reply]
OK, kicking myself now... found C:\bin]pwsh.exe that's a symlink to pwsh.exe that's in the path... deleted the symlink & it works... glad I found that before I filed a bug report... disappointed I list almost a full day of work trying to chase that down
